clang/test/Driver/sycl-offload.c

```diff
@@ -1138,7 +1137,7 @@
 // CHK-INCOMPATIBILITY: error: The option -fsycl conflicts with -ffreestanding
 
 /// Using -fsyntax-only with -fsycl should not emit IR
-// RUN:   %clang -### -fsycl -fsyntax-only %s 2>&1 \
+// RUN:   %clang -### -target x86_64-unknown-linux-gnu -fsycl -fsyntax-only %s 2>&1 \
```

Comment:
Could you clarify why we need `-target` option?

Reply:
It essentially overrides the driver-detected host environment with the provided host triple (since the patch aims to [work around the Win32 failures](https://github.com/intel/llvm/pull/4962#issuecomment-969736320)).

Reply:
> It essentially overrides the driver-detected host environment with the provided host triple

The test checks options passed to device compiler, so it's not clear why this change is needed.
I don't expect any changes in test behavior here. What do I miss?

Reply:
After https://github.com/intel/llvm/pull/4936, the 32-bit SPIR triple is enforced for 32-bit host systems, which I assume breaks the Win32 checks for line 1142 of the new version. However, I agree that extending the device triple check to account for that could also be an option. @mdtoguchi, could you please comment on why the latter method is not as well-applicable here?

Reply:
Maybe `spir{{.*}}-unknown-unknown` could be used in the check line as a compromise?

Reply:
Thanks for the comments @bader, I have simplified the test.  We have the additional check for not `-emit-llvm-bc` due to the fact that `-fsycl` does 2 compilations.